### PR TITLE
Disallow setting editor-only metadata in the editor

### LINKED ISSUE #59852: Metadata editor allows defining editor-only metadata
### Godot version

4.0.alpha (c630c2001)

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 510.60.02)

### Issue description

The [new metadata editor](https://github.com/godotengine/godot/pull/59452) allows defining editor-only metadata, but not reading it. This is probably confusing and should be disallowed (although setting editor-only metadata from code should remain allowed).

### Steps to reproduce

- Add a node.
- Select the node in the Scene tree dock, go to the inspector and click **Add Metadata** at the bottom.
- Add metadata with the name `_test`.
- Notice that the metadata doesn't appear in the inspector.
- Attach a script to the node, with `print(has_meta("_test"))` in the `_ready()` function.
- Run the project. Notice that it prints `true` since the metadata *is* defined – it's just not visible in the editor.

### Minimal reproduction project

[test_editor_only_metadata.zip](https://github.com/godotengine/godot/files/8405566/test_editor_only_metadata.zip)

### PR DESCRIPTION
**Pull request by RechieKho**
Issue: https://github.com/godotengine/godot/issues/59852

NotAllow

Metadata starts with `_` considered as editor-only metadata and disallowed as mentioned in the issue. 

A bit sorry for [Ayush-singla27](https://github.com/Ayush-singla27), it has been 3 days and it really just need a few lines of code (or maybe Ayush already make a PR and I don't know how to use github). This is, too, my first contribution to me beloved game engine.

<i>Bugsquad edit:</i>
- Fix #59852

### CODE REVIEW

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -3720,6 +3720,8 @@ void EditorInspector::_check_meta_name(String name) {
 		error = TTR("Invalid metadata identifier.");
 	} else if (object->has_meta(name)) {
 		error = TTR("Metadata already exists.");
+	} else if (name[0] == '_') {
+		error = TTR("Editor-only metadata only allowed setting in code.");
```
Comment (Calinou) [inline editor/editor_inspector.cpp:3724]: Use a more informative error message:

```suggestion
		error = TTR("Names starting with _ are reserved for editor-only metadata.");
```

**Review by Calinou — CHANGES_REQUESTED** [review]

**Comment by RechieKho** [comment]
Commited

**Review by Calinou — APPROVED** [review]
Looks good to me, please squash the commits together so we can merge this. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions :slightly_smiling_face:

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**